judge/admin/submission.py

```diff
@@ -244,7 +246,9 @@ def language_column(self, obj):
 
     @admin.display(description='')
     def judge_column(self, obj):
-        if obj.is_locked:
+        if obj.is_archived:
+            return format_html('<input type="button" disabled value="{0}"/>', _('Archived'))
+        elif obj.is_locked:
```

Comment:
As discussed in discord briefly, a submission being archived but not locked should mean "this submission should be invalidated for points, but is otherwise valid". So maybe archived submissions should still be rejudgeable?